### PR TITLE
🐛 Add pip installs to Pinned-Dependencies score

### PR DESCRIPTION
**Pull request by gabibguti**
#### What kind of change does this PR introduce?

This is a bug fix. Unpinned pip installs were identified but did not count for Pinned-Dependencies score. Now, they count.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

Pip installs are identified as pinned or unpinned and reported by Scorecard, but do not count for Pinned-Dependencies score.

#### What is the new behavior (if this is a feature change)?**

Pip installs now count in the aggregated score for Pinned-Dependencies.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

Fixes https://github.com/ossf/scorecard/issues/2908

Plus, related to https://github.com/ossf/scorecard/issues/2582

#### Special notes for your reviewer

@raghavkaul This PR addresses one of the changes made in https://github.com/ossf/scorecard/pull/2690.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Pip installs count for Pinned-Dependencies score.
```

### CODE REVIEW

File: checks/evaluation/pinned_dependencies.go [hunk 1]
```diff
@@ -121,14 +121,21 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
+	// Pip installs.
+	pipScore, err := createReturnForIsPipInstallPinned(pr, dl)
+	if err != nil {
```
Comment (naveensrinivasan) [inline checks/evaluation/pinned_dependencies.go:126]: Thanks! Can you also please include tests for this?

**Review by naveensrinivasan — APPROVED** [review]
Thanks!

File: checks/evaluation/pinned_dependencies.go (hunk 1 above)
Comment (gabibguti) [reply checks/evaluation/pinned_dependencies.go:126]: Yes, sure! Thanks for reviewing Naveen!

File: checks/evaluation/pinned_dependencies.go [hunk 2]
```diff
@@ -121,14 +121,21 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
+	// Pip installs.
+	pipScore, err := createReturnForIsPipInstallPinned(pr, dl)
```
Comment (laurentsimon) [inline checks/evaluation/pinned_dependencies.go:125]: what happens to the other Npm, curl, etc? Are they also not accounted for in the final score?

Comment (gabibguti) [reply checks/evaluation/pinned_dependencies.go:125]: Yes, they are also not accounted in the final score. I plan on including npm, choco and go commands in separate PRs to count for the final score. curl is a different case, it already counts in the final score.

Comment (laurentsimon) [reply checks/evaluation/pinned_dependencies.go:125]: SG, thanks @gabibguti

File: checks/evaluation/pinned_dependencies.go (hunk 1 above)
Comment (gabibguti) [reply checks/evaluation/pinned_dependencies.go:126]: @naveensrinivasan I added some test cases! I don't know if it's possible to test the case where a dependency ends in "Inconclusive score: -1" or in error. Let me know what you think.

Comment (naveensrinivasan) [reply checks/evaluation/pinned_dependencies.go:126]: Sounds good. Thanks

**Pull request merged**